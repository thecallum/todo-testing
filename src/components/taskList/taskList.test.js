import { render, screen, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import TaskList from "./taskList";

describe("TaskList", () => {
  it("Renders component when many tasks", () => {
    // Arrange
    const tasks = [
      { value: "task description one", completed: false },
      { value: "task description two", completed: true },
    ];
    const toggleCheckbox = () => {};
    const deleteTask = () => {};

    // Act
    const { asFragment } = render(
      <TaskList
        tasks={tasks}
        toggleCheckbox={toggleCheckbox}
        deleteTask={deleteTask}
      />
    );

    // Assert
    expect(asFragment()).toMatchSnapshot();
  });

  it("Renders component when no tasks", () => {
    // Arrange
    const tasks = [];
    const toggleCheckbox = () => {};
    const deleteTask = () => {};

    // Act
    const { asFragment } = render(
      <TaskList
        tasks={tasks}
        toggleCheckbox={toggleCheckbox}
        deleteTask={deleteTask}
      />
    );

    // Assert
    expect(asFragment()).toMatchSnapshot();
  });
});

