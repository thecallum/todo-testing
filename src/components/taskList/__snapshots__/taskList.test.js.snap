// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList Renders component when many tasks 1`] = `
<DocumentFragment>
  <ul
    class="tasklist"
  >
    <li
      class="tasklist-item"
    >
      <div
        class="tasklist-item-left"
      >
        <input
          class="tasklist-item-input"
          type="checkbox"
        />
        <span
          class="tasklist-item-value "
        >
          task description one
        </span>
      </div>
      <button
        aria-label="Delete task"
        class="tasklist-item-button"
      />
    </li>
    <li
      class="tasklist-item"
    >
      <div
        class="tasklist-item-left"
      >
        <input
          checked=""
          class="tasklist-item-input"
          type="checkbox"
        />
        <span
          class="tasklist-item-value completed"
        >
          task description two
        </span>
      </div>
      <button
        aria-label="Delete task"
        class="tasklist-item-button"
      />
    </li>
  </ul>
</DocumentFragment>
`;

exports[`TaskList Renders component when no tasks 1`] = `
<DocumentFragment>
  <div
    class="tasklist-empty"
  >
    You have no tasks
  </div>
</DocumentFragment>
`;
